### PR TITLE
hashi-ui: init at 1.3.8

### PR DESCRIPTION
**Pull request by numkem**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/hashi-ui/default.nix
Comment (cole-h) [inline pkgs/servers/hashi-ui/default.nix:1]:
```suggestion
{ stdenv, fetchurl }:
```

File: pkgs/servers/hashi-ui/default.nix
Comment (cole-h) [inline pkgs/servers/hashi-ui/default.nix:20]:
```suggestion
    homepage = "https://github.com/jippi/hashi-ui";
```

File: pkgs/servers/hashi-ui/default.nix
Comment (cole-h) [inline pkgs/servers/hashi-ui/default.nix:23]: You need to actually add yourself to the [`maintainer-list.nix` file](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) (in a separate commit, please: `maintainers: add numkem`).

File: pkgs/top-level/all-packages.nix
Comment (cole-h) [inline pkgs/top-level/all-packages.nix:26451]: Since this appears to be a server, maybe it would be better served (ha) under the servers header (alphabetically, please): https://github.com/NixOS/nixpkgs/blob/045384ac98ced4f34b735a9dc56c974759d2d52c/pkgs/top-level/all-packages.nix#L15299

All headers start with `###`, so keep it between `### SERVERS` and `### OS-SPECIFIC`: https://github.com/NixOS/nixpkgs/blob/045384ac98ced4f34b735a9dc56c974759d2d52c/pkgs/top-level/all-packages.nix#L6

**Review by cole-h — CHANGES_REQUESTED** [review]
Is there a reason why you opted for just fetching the binary instead of building it from source? I don't speak with any authority, but I think it's preferable to build from source wherever possible.

**Comment by numkem** [comment]
@cole-h much appreciated for your time. It's my first package and the learning curve is... Curved.

To answer your question about building from source instead. I tried but it requires some yarn packages that aren't in nixpkgs already and the `Makefile` installs them anyway and gave me errors.  While I know this isn't the best approach, it is working and the build is "official" as it's straight from github releases.

**Review by cole-h — CHANGES_REQUESTED** [review]
I don't like it, but it's understandable. Unless someone else considers that a problem, should be good to go (after you put your maintainer stuff where it belongs, alphabetically).

File: maintainers/maintainer-list.nix
Comment (numkem) [reply maintainers/maintainer-list.nix]: My bad, only read the bottom of the file as there are some entries that aren't in order.

Comment (cole-h) [reply maintainers/maintainer-list.nix]: Sometimes, people just don't pay attention/overlook that when reviewing ;) Thanks for fixing it.

**Review by cole-h — APPROVED** [review]
Assuming nobody else comes along and requests additional changes, LGTM. Binary runs, displays some info, and exits (because it wants Consul and/or Nomad, which I don't have).

```
https://github.com/NixOS/nixpkgs/pull/84286
20 package marked as broken and skipped:
clang-sierraHack digitalbitbox glibcCross iterm2 linuxPackages_4_4.evdi linuxPackages_hardkernel_4_14.bcc linuxPackages_hardkernel_4_14.bpftrace linuxPackages_hardkernel_4_14.can-isotp linuxPackages_hardkernel_4_14.digimend linuxPackages_hardkernel_4_14.evdi linuxPackages_hardkernel_4_14.mba6x_bl linuxPackages_hardkernel_4_14.nvidia_x11 linuxPackages_hardkernel_4_14.nvidia_x11_legacy390 linuxPackages_hardkernel_4_14.nvidiabl php74Packages.pcs python27Packages.caffe python27Packages.flitBuildHook python37Packages.nixpart python38Packages.libselinux python38Packages.nixpart

1 package built:
hashi-ui
```

**Comment by DamienCassou** [comment]
Your PR contains merge commits. Please get rid of them and rebase on top of latest `master`.

**Comment by numkem** [comment]
@DamienCassou this should do it!

File: pkgs/servers/hashi-ui/default.nix
Comment (cole-h) [inline pkgs/servers/hashi-ui/default.nix:22]: This should probably be:

```suggestion
    platforms = [ "x86_64-linux" ];
```

since upstream only ships 64 bit releases.

**Pull request merged**
